src/manifolds/ProbabilitySimplex.jl

```diff
@@ -0,0 +1,373 @@
+i.e. the distanceradius to a point near/on the boundary, that could be reached by following the
+geodesic.
+"""
+function injectivity_radius(::ProbabilitySimplex{n}, p) where {n}
+    i = argmin(p)
+    s = sum(p) - p[i]
+    return 2 * acos(sqrt(s))
+end
+function injectivity_radius(M::ProbabilitySimplex, p, ::ExponentialRetraction)
+    return injectivity_radius(M, p)
+end
+injectivity_radius(M::ProbabilitySimplex, p, ::SoftmaxRetraction) = injectivity_radius(M, p)
+injectivity_radius(M::ProbabilitySimplex) = 0
+injectivity_radius(M::ProbabilitySimplex, ::SoftmaxRetraction) = 0
+injectivity_radius(M::ProbabilitySimplex, ::ExponentialRetraction) = 0
+eval(
+    quote
+        @invoke_maker 1 Manifold injectivity_radius(
+            M::ProbabilitySimplex,
+            rm::AbstractRetractionMethod,
+        )
+    end,
+)
+
+@doc raw"""
+    inner(M::ProbabilitySimplex,p,X,Y)
+
+Compute the inner product of two tangent vectors `X`, `Y` from the tangent space $T_pΔ^n$ at
+`p`. The formula reads
+````math
+g_p(X,Y) = \sum_{i=1}^{n+1}\frac{X_iY_i}{p}
+````
+"""
+function inner(::ProbabilitySimplex, p, X, Y)
+    d = zero(Base.promote_eltype(p, X, Y))
+    @inbounds for i in eachindex(p, X, Y)
+        d += X[i] * Y[i] / p[i]
+    end
+    return d
+end
+
+@doc raw"""
+    inverse_retract(M::ProbabilitySimplex, p, q, ::SoftmaxInverseRetraction)
+
+Compute a first order approximation by projection. The formula reads
+````math
+\operatorname{retr}^{-1}_p q = \bigl( I - \frac{1}{n}\mathbb{1}\mathbb{1}^\mathrm{T}\bigr)(\log(q)-\log(p))
+````
+where $\mathbb{1}$ is the column vector containing ones and $\log$ is applied elementwise.
+"""
+inverse_retract(::ProbabilitySimplex, ::Any, ::Any, ::SoftmaxInverseRetraction)
+
+function inverse_retract!(
+    ::ProbabilitySimplex{n},
+    X,
+    p,
+    q,
+    ::SoftmaxInverseRetraction,
+) where {n}
+    X .= log.(q) .- log.(p)
+    meanlogdiff = mean(X)
+    X .-= meanlogdiff
+    return X
+end
+
+
+@doc raw"""
+    log(M::ProbabilitySimplex, p, q)
+
+Compute the logarithmic map of `p` and `q` on the [`ProbabilitySimplex`](@ref) `M`.
+
+````math
+\log_pq = \frac{d_{Δ^n}(p,q)}{\sqrt{1-⟨\sqrt{p},\sqrt{q}⟩}}(\sqrt{pq} - ⟨\sqrt{p},\sqrt{q}⟩p),
+````
+
+where $pq$ and $\sqrt{p}$ is meant elementwise.
+"""
+log(::ProbabilitySimplex, ::Any...)
+
+function log!(M::ProbabilitySimplex, X, p, q)
+    if p ≈ q
+        fill!(X, 0)
+    else
+        z = sqrt.(p .* q)
+        s = sum(z)
+        X .= 2 * acos(s) / sqrt(1 - s^2) .* (z .- s .* p)
+    end
+    return X
+end
+
+@doc raw"""
+    manifold_dimension(M::ProbabilitySimplex{n})
+
+Returns the manifold dimension of the probability simplex in $ℝ^{n+1}$, i.e.
+````math
+    \dim_{Δ^n} = n.
+````
+"""
+manifold_dimension(::ProbabilitySimplex{n}) where {n} = n
+
+@doc raw"""
+    mean(
+        S::ProbabilitySimplex,
+        x::AbstractVector,
+        [w::AbstractWeights,]
+        method = GeodesicInterpolationWithinRadius();
```

Comment:
So I _still_ think that we can just use `GeodesicInterpolation` here, as the entire simplex is included in the corresponding radius for the sphere of radius 2, unless there's a good reason not to?

Reply:
Oh, I misread your last request on that, will change this to the right form then

Reply:
Thanks! Yeah, just plain `GeodesicInterpolation()` everywhere should do it.